src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -78,6 +85,27 @@ public JavaGapicSurfaceTestTransformer(GapicCodePathMapper javaPathMapper) {
     return views;
   }
 
+  public Iterable<Interface> getGrpcInterfacesToMock(Model model, ApiConfig apiConfig) {
+    Map<String, Interface> interfaces = new LinkedHashMap<>();
+
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      if (!service.isReachable()) {
+        continue;
+      }
+      interfaces.put(service.getFullName(), service);
+      InterfaceConfig interfaceConfig = apiConfig.getInterfaceConfig(service);
+      for (MethodConfig methodConfig : interfaceConfig.getMethodConfigs()) {
+        String reroute = methodConfig.getRerouteToGrpcInterface();
+        if (!Strings.isNullOrEmpty(reroute)) {
+          Interface targetInterface = model.getSymbolTable().lookupInterface(reroute);
+          interfaces.put(reroute, targetInterface);
+        }
+      }
+    }
+
+    return interfaces.values();
```

Comment:
That won't happen if my logic is correct. I put Interface instances into the interfaces map keyed by the full name of the interface. So, if an interface is encountered more than once, it will occur in the result only once.